packages/sitecore-jss/src/layout/models.ts

```diff
@@ -44,7 +44,7 @@ export interface RouteData {
   name: string;
   displayName?: string;
   fields?: {
-    [name: string]: Field;
+    [name: string]: Field | Item | Item[];
```

Comment:
I am wondering if we could go 1 step further and make `RouteData` a generic interface like so:

```ts
export interface RouteData<T = Record<string, Field | Item | Item[]>> {
  fields?: T
 // ...rest
}
```

Reply:
@ambrauer What do you think about generic type for fields? I decided to add, I agree with @pzi, can be helpful

Reply:
I like it 👍

Reply:
![](https://gifs.pzi.io/high-five/it-crowd-high-five.gif)